packages/lit-element/src/lib/lit-element.ts

```diff
@@ -126,58 +125,42 @@ export class LitElement extends UpdatingElement {
    * using the [[`css`]] tag function or via constructible stylesheets.
    */
   static styles?: CSSResultOrNative | CSSResultArray;
-
-  private static _styles: Array<CSSResultOrNative | CSSResult> | undefined;
+  private static elementStyles?: CSSResultArray;
 
   /**
-   * Return the array of styles to apply to the element.
+   * Takes the styles the user supplied via the `static styles` property and
+   * returns the array of styles to apply to the element.
    * Override this method to integrate into a style management system.
    *
+   * Styles are deduplicated preserving the _last_ instance in the list. This
+   * is a performance optimization to avoid duplicated styles that can occur
+   * especially when composing via subclassing. The last item is kept to try
+   * to preserve the cascade order with the assumption that it's most important
+   * that last added styles override previous styles.
+   *
    * @nocollapse
    */
-  static getStyles(): CSSResultOrNative | CSSResultArray | undefined {
-    return this.styles;
-  }
-
-  /** @nocollapse */
-  private static _getUniqueStyles() {
-    // Only gather styles once per class
-    if (this.hasOwnProperty(JSCompiler_renameProperty('_styles', this))) {
-      return;
-    }
-    // Take care not to call `this.getStyles()` multiple times since this
-    // generates new CSSResults each time.
-    // TODO(sorvell): Since we do not cache CSSResults by input, any
-    // shared styles will generate new stylesheet objects, which is wasteful.
-    // This should be addressed when a browser ships constructable
-    // stylesheets.
-    const userStyles = this.getStyles();
-
-    if (Array.isArray(userStyles)) {
-      // De-duplicate styles preserving the _last_ instance in the set.
-      // This is a performance optimization to avoid duplicated styles that can
-      // occur especially when composing via subclassing.
-      // The last item is kept to try to preserve the cascade order with the
-      // assumption that it's most important that last added styles override
-      // previous styles.
+  protected static getStyles(
+    styles?: CSSResultOrNative | CSSResultArray
+  ): CSSResultArray {
+    const elementStyles = [];
+    if (Array.isArray(styles)) {
       const addStyles = (
-        styles: CSSResultArray,
+        stylesList: CSSResultArray,
         set: Set<CSSResultOrNative>
       ): Set<CSSResultOrNative> =>
-        styles.reduceRight(
+        stylesList.reduceRight(
```

Comment:
Can you use `new Set(stylesList.flat(Infinity))`

Reply:
Sheesh, yes I believe so. We need to reverse the flattened array so that the set preserves the last instance. I believe we didn't do this initially because IE doesn't have Array.flat.

Reply:
Yeah, I'm not sure all evergreen browsers had it then and this was smaller than a polyfill. But now I think we can go for it.